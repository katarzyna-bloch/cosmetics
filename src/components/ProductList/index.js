import styled from 'styled-components'
import PropTypes from 'prop-types'
import React from 'react'

import Product from '../Product'

const List = styled.div`
  display: flex;
  flex: 1;
  flex-wrap: wrap;
`

const ProductList = ({ products }) => ( 
  <List>
    {products.map(({ name, img }) => (
      <Product
        key={name}
        name={name}
        img={img}
      />
    ))}
  </List>
)

ProductList.propTypes = {
  products: PropTypes.arrayOf(PropTypes.shape({
    name: PropTypes.string.isRequired,
    img: PropTypes.string.isRequired,
  })).isRequired,
}

export default ProductList
