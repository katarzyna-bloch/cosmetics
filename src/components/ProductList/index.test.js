import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import ProductList from '.'

it('should render correctly', () => {
  const props = {
    products: ['Test', 'img']
    }
  const wrapped = shallow(<ProductList {...props} />)
  expect(toJson(wrapped)).toMatchSnapshot()
})
