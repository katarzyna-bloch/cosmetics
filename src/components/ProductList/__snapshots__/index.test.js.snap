// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly 1`] = `
<styled.div>
  <Product />
  <Product />
</styled.div>
`;
